qutebrowser/utils/urlutils.py

```diff
@@ -176,7 +179,7 @@ def fuzzy_url(urlstr, cwd=None, relative=False, do_search=True):
 
     if path is not None:
         url = QUrl.fromLocalFile(path)
-    elif (not do_search) or is_url(urlstr):
+    elif not force_search and (not do_search or is_url(urlstr)):
```

Comment:
I think it's time to rewrite this `if` so that the search term is handled first (`elif (do_search and is_url(urlstr)) or force_search:` I think?) and handle the address case in `else:` - otherwise this gets kind of hard to read.